data/helpers.d/filesystem

```diff
@@ -80,3 +80,49 @@ properly with chmod/chown." >&2
     chmod 755 $TMP_DIR
     echo $TMP_DIR
 }
+
+# Calculate and store a file checksum into the app settings
+#
+# $app should be defined when calling this helper
+#
+# usage: ynh_store_file_checksum file
+# | arg: file - The file on which the checksum will performed, then stored.
+ynh_store_file_checksum () {
+	local checksum_setting_name=checksum_${1//[\/ ]/_}	# Replace all '/' and ' ' by '_'
+	ynh_app_setting_set $app $checksum_setting_name $(sudo md5sum "$1" | cut -d' ' -f1)
+}
+
+# Verify the checksum and backup the file if it's different
+# This helper is primarily meant to allow to easily backup personalised/manually 
+# modified config files.
+#
+# $app should be defined when calling this helper
+#
+# usage: ynh_backup_if_checksum_is_different file [compress]
+# | arg: file - The file on which the checksum test will be perfomed.
+# | arg: compress - 1 to compress the backup instead of a simple copy
+# A compression is needed for a file which will be analyzed even if its name is different.
```

Comment:
Indeed, thanks! I think we can afford an additional line of comment to add this explanation, because without it it was really not clear for me.